blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py

```diff
@@ -65,8 +65,10 @@ def parse_page_token(data, url):
   page_token = data.get('nextPageToken')
   if page_token:
     logging.info(f'page  token {page_token}')
-  if page_token:
-    return RE_URL.sub(f'pageToken={page_token}&', url)
+    if 'pageToken' in url:
+      return RE_URL.sub(f'pageToken={page_token}', url)
+    else:
+      return f'{url}&pageToken={page_token}'
```

Comment:
```suggestion
    parsed_url = urllib.parse.urlparse(url)
    parsed_qs = urllib.parse.parse_qs(parsed_url.query)
    parsed_qs['pageToken'] = page_token
    return parsed_url._replace(query=urllib.parse.urlencode(parsed_qs, doseq=True)).geturl()
```

And then you can get rid of `RE_URL`

Reply:
I find yours harder to read Wiktor :)

Reply:
Yes, it is a bit more complex, but handles all the edge cases.

`RE_URL.sub` will work for now, so I'm open to reject my suggestion.